### DISCUSSION TITLE
Error 413 (Request Entity Too Large) When Using "Select Recent Configurations" in Airflow UI

### DISCUSSION THREAD

**Discussion by Glitch33r** [apache/airflow#45199] [Q&A] [answered]
### Description:
I am encountering an issue in the Airflow UI when attempting to run a DAG using the "Select Recent Configurations" option. Instead of running successfully, I receive a 413 Request Entity Too Large error.

**JSON Size in Issue:** 10,000+ lines

### Steps to Reproduce:
  - Open the form to manually provide JSON data for the DAG run (our JSON can be quite large, up to 10,000 lines).
  - Submit the JSON configuration. This step works perfectly, and the DAG runs without issues.
  - Attempt to use "Select Recent Configurations" to reuse a previously submitted JSON configuration.
  - Encounter the 413 Request Entity Too Large error.

### Environment:
**Airflow Version**: 2.10.3

### Additional Information:
The error suggests that the payload is too large for the server to handle. However, submitting the same data via the form works perfectly.

### Questions:

  - Is this a known limitation or bug in the Airflow UI?
  - Are there any configuration options (e.g., for the web server) that can address this issue?
  - Could this be resolved by modifying request limits in the server configuration?

Any help or insights would be greatly appreciated! 

<img width="388" alt="image" src="https://github.com/user-attachments/assets/d7da97d6-4463-4f75-9b2d-d25323a22c13" />

**Comment by potiuk** [comment, marked as answer]
https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#allowed-payload-size

**Reply by Glitch33r** [reply]
Thank you! I missed this conf variable

**Reply by HuanjieGuo** [reply]
Hi @Glitch33r , does it work for you after you tune the payload size? For our case, it doesn't work even after we tuned it.

**Reply by Glitch33r** [reply]
@HuanjieGuo Hi, yes. It works for my docker instance and on production as well. Maybe try as env variable (I have done in this way), also try to increase the value and check if this particular config is used

**Reply by HuanjieGuo** [reply]
Thanks @Glitch33r , let me change it to env variable. Currently we use airflow config.

**Reply by HuanjieGuo** [reply]
sys env doesn't help in 2.10.5 either.

**Comment by HuanjieGuo** [comment]
Same here, use Airflow 2.10.5

run a DAG using the "Select Recent Configurations" option. Instead of running successfully, I receive a 413 Request Entity Too Large error.

Already set the [allowed-payload-size](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#allowed-payload-size) and it doesn't work .

<img width="775" height="288" alt="image" src="https://github.com/user-attachments/assets/d2f6b7b4-194a-4a1d-aa97-fed964c0dc7c" />

The payload only takes around 111 kb.

<img width="751" height="35" alt="image" src="https://github.com/user-attachments/assets/6049962b-20eb-4609-a922-f7abf0783596" />

**Reply by potiuk** [reply]
Actually the 111KB is likely the HTML page with "Request Entity too large' message - not the actual payload. If you received 413 error it means that you DID NOT get what you wanted because it was too large.

Also it could be your proxy or whatever you have between you and airflow that might cause it.

**Reply by HuanjieGuo** [reply]
We deployed our cluster in k8s and the request has passed the service. 

"Actually the 111KB is likely the HTML page with "Request Entity too large' message" if it is from HTML, that should due to the webserver code right? 

It's weird that tune the allowed_payload_size doesn't work for it. Any other parameter we can try for the html size you think?
